src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/OrderIT.java

```diff
@@ -0,0 +1,168 @@
+        JSONArray result = getSortExplain(query);
+        Assert.assertThat(1, equalTo(result.length()));
+        JSONObject jsonObject = getSortByField(result, "id");
+        Assert.assertThat("asc", equalTo(getOrderTypeForSortAtIdex(result, 0, "id")));
+        Assert.assertFalse(jsonObject.getJSONObject("id").has("missing"));
+    }
+
+    @Test
+    public void orderByScore() throws IOException {
+        String query = "SELECT * FROM elasticsearch-sql_test_index_order ORDER BY _score";
+        JSONArray result = getSortExplain(query);
+        Assert.assertThat(1, equalTo(result.length()));
+        JSONObject jsonObject = getSortByField(result, "_score");
+        Assert.assertThat("asc", equalTo(getOrderTypeForSortAtIdex(result, 0, "_score")));
+        Assert.assertFalse(jsonObject.getJSONObject("_score").has("missing"));
+
+        JSONObject response = executeQuery(query);
+        JSONArray hits = getHits(response);
+        Assert.assertThat(7, equalTo(hits.length()));
+    }
+
+    @Test
+    public void simpleOrderMultipleFields() throws IOException {
+        String query = "SELECT id, name FROM elasticsearch-sql_test_index_order ORDER BY id, name";
+        JSONArray result = getSortExplain(query);
+        Assert.assertThat(2, equalTo(result.length()));
+        Assert.assertTrue(result.getJSONObject(0).has("id"));
+        Assert.assertTrue(result.getJSONObject(1).has("name.keyword"));
+    }
+
+    @Test
+    public void explicitOrderType() throws IOException {
+        String query = "SELECT id, name FROM elasticsearch-sql_test_index_order ORDER BY id ASC, name DESC";
+        JSONArray result = getSortExplain(query);
+        Assert.assertThat(2, equalTo(result.length()));
+        Assert.assertThat("asc", equalTo(getOrderTypeForSortAtIdex(result, 0, "id")));
+        Assert.assertThat("desc", equalTo(getOrderTypeForSortAtIdex(result, 1, "name.keyword")));
+    }
+
+    @Test
+    public void orderByIsNull() throws IOException {
+        String query = "SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL, id DESC";
+        JSONArray result = getSortExplain(query);
+        String explain_one = explainQuery(query);
+        Assert.assertThat(1, equalTo(result.length()));
+        Assert.assertThat("desc", equalTo(getOrderTypeForSortAtIdex(result, 0, "id")));
+        Assert.assertThat(
+            "_last", equalTo((result.getJSONObject(0)).getJSONObject("id").getString("missing"))
+        );
+
+        JSONObject response = executeQuery(query);
+        JSONArray hits = getHits(response);
+        Assert.assertThat(5, equalTo(hits.getJSONObject(0).getJSONObject("_source").getInt("id")));
+
+        // Another equivalent syntax
+        query = "SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL DESC";
+        String explain_two = explainQuery(query);
+        Assert.assertThat(explain_one, equalTo(explain_two));
```

Comment:
Is this part integration test though?

Feels like it could be extracted to separate unit test (explain doesn't do server calls, so this is not necessary to be integration test)

assertThat(explainQuery("SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL, id DESC"), equalTo(explainQuery("SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL DESC"));

Reply:
The `explainQuery` actually makes server calls (uses `RestClient` underneath). I would like to keep this as integration test.


Will change the assert format as you described.
```
@Test
someTest() {
assertThat(explainQuery("SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL, id DESC"), equalTo(explainQuery("SELECT * FROM elasticsearch-sql_test_index_order ORDER BY id IS NULL DESC"));
}
```